packages/tasks_panel/src/multi-step-select.ts

```diff
@@ -24,32 +25,52 @@ function grabProjectItems(tasks: ConfiguredTask[], project?: string): QuickPickI
   });
 }
 
-async function grabTasksByGroup(tasks: ConfiguredTask[], project: string): Promise<QuickPickItem[]> {
-  function toFioriE2ePickItems(items: FioriProjectInfo[]): QuickPickItem[] {
+async function grabTasksByGroup(
+  tasks: ConfiguredTask[],
+  project: string,
+  group?: string | undefined
+): Promise<QuickPickItem[]> {
+  function toConfigDeployE2ePickItems(items: ProjectConfigInfo[]): QuickPickItem[] {
     return map(items, (item) => {
       return {
-        label: `Define Deployment parameters - ${item.project || item.wsFolder}`,
+        label: "Define Deployment parameters",
+        detail: `${item.project || item.wsFolder}`,
         ...item,
       };
     });
   }
+
   const pickItems: any[] = [];
-  const fioriDeploymentParamItems = toFioriE2ePickItems(await getFioriE2ePickItems(project));
-  if (!isEmpty(fioriDeploymentParamItems)) {
-    pickItems.push({ label: "Fiori Configuration", kind: QuickPickItemKind.Separator });
-    pickItems.push(...fioriDeploymentParamItems);
+  const deploymentParamItems = !group ? toConfigDeployE2ePickItems(await getConfigDeployPickItems(project)) : [];
+  if (!isEmpty(deploymentParamItems)) {
+    const groupByType = groupBy(deploymentParamItems, "type");
+    each(keys(groupByType), (key) => {
+      pickItems.push({ label: composeDeploymentConfigLabel(key), kind: QuickPickItemKind.Separator });
+      pickItems.push(...groupByType[key]);
+    });
   }
+
   const tasksByProject = filter(tasks, ["__wsFolder", project]);
-  each(sortBy(uniq(map(tasksByProject, "__intent"))), (intent) => {
+  each(sortBy(uniq(map(tasksByProject, "__intent"))), (intent: string) => {
     // add a group separator
     if (isMatchDeploy(intent) || isMatchBuild(intent)) {
-      if (isEmpty(fioriDeploymentParamItems)) {
-        // fiori projects workaround: hide `Build`/`Deploy` npm tasks if fiori e2e deployment configuration needed
-        pickItems.push({ label: intent, kind: QuickPickItemKind.Separator });
-        pickItems.push(...filter(tasksByProject, ["__intent", intent]));
+      if (isEmpty(deploymentParamItems)) {
+        if (!group || group.toLowerCase() === intent.toLowerCase()) {
+          // fiori projects workaround: hide `Build`/`Deploy` npm tasks if fiori e2e deployment configuration needed
+          pickItems.push({ label: intent, kind: QuickPickItemKind.Separator });
+          pickItems.push(
+            ...compact(
+              map(tasksByProject, (_) => {
```

Comment:
maybe too many operations in one expression?

instead of combining the mapping and filtering (via compact)
1. First filter all those with relevant `intent`
2. Secondly map this sub-array to the new structure